src/subdomains/core/payment-link/services/payment-quote.service.ts

```diff
@@ -53,6 +59,31 @@ export class PaymentQuoteService {
     }
   }
 
+  async processTransactions(): Promise<void> {
+    const blockchainPaymentLinkQuotes = await this.paymentQuoteRepo.find({
+      where: {
+        status: PaymentQuoteStatus.TX_BLOCKCHAIN,
+        activations: { status: PaymentActivationStatus.COMPLETED },
+      },
+      relations: { activations: true },
+    });
+
+    for (const blockchainPaymentLinkQuote of blockchainPaymentLinkQuotes) {
+      const blockchain = blockchainPaymentLinkQuote.activations[0]?.method;
+
+      if (blockchain) {
+        const client = this.evmRegistryService.getClient(blockchain);
+        const transaction = await client.getTransaction(blockchainPaymentLinkQuote.txId);
+
+        const minNumberOfConfirmations = blockchain === Blockchain.ETHEREUM ? 6 : 100;
+
+        if (transaction?.confirmations >= minNumberOfConfirmations) {
+          await this.saveFinallyConfirmed(blockchainPaymentLinkQuote.uniqueId);
+        }

```

Comment:
Maybe use `EvmClient.isTxComplete` and add an optional `confirmations` parameter?